### ISSUE TITLE
MultilineOperationIndentation not working properly on implicit return

### ISSUE THREAD

**Issue by ngouy** [rubocop/rubocop#10345]
## disclamer

I'm not sure if it's a bug or not.

--------

Per the [docs](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Layout/MultilineOperationIndentation):
>The `aligned` style checks that operators are aligned if they are part of an `if` or `while` condition, a `return` statement, etc. In other contexts, the second operand should be indented regardless of enforced style.

That's not always true, at least for explicit returns

Let's have the following in config file:

```yml
# just for the sake of this example
Style/RedundantReturn:
  Enabled: false

Layout/MultilineOperationIndentation:
  EnforcedStyle: aligned
```

And the following code
```rb
def my_test
  my_var >>
  my_function >>
  my_other_function
end

def another_test
  return my_var >>
         my_function >>
         my_other_function
end
```

## Expected behavior
The 2 examples should (?) be ok

## Actual behavior

In reality, the first one is failing, as (my guess) it's not "detecting" that this really is a return statement.

## RuboCop version

```
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.0, running on ruby 2.7.5 x86_64-darwin20)
  - rubocop-faker 1.1.0
  - rubocop-performance 1.12.0
  - rubocop-rails 2.12.4
```

**Comment by ngouy** [comment]
cc @jonas054 (sorry for the ping, but since you are owner of this, maybe you could first assess if it's bug, or if it's meant to work like that)

**Comment by jonas054** [comment]
Yes, that's intended. It's generally speaking not possible for RuboCop to figure out what's an implicit return and what's a regular statement. I'll clarify that in the docs by adding "explicit" in front of "return". I also noticed that the examples given in the docs are incorrect, so I'll update those as well.

**Comment by ngouy** [comment]
🙏 thanks for the update
